lib/iris/cube.py

```diff
@@ -991,6 +991,157 @@ def replace_coord(self, new_coord):
         for factory in self.aux_factories:
             factory.update(old_coord, new_coord)
 
+    def promote_aux_coord_to_dim_coord(self, name_or_coord):
+        """
+        Promotes an AuxCoord on the cube to a DimCoord. This AuxCoord must be
+        associated with a single cube dimension. If the AuxCoord is associated
+        with a dimension that already has a DimCoord, that DimCoord gets
+        demoted to an AuxCoord.
+
+        Args:
+
+        * name_or_coord:
+            Either
+
+            (a) An instance of :class:`iris.coords.AuxCoord`
+
+            or
+
+            (b) the :attr:`standard_name`, :attr:`long_name`, or
+            :attr:`var_name` of an instance of an instance of
+            :class:`iris.coords.AuxCoord`.
+
+        For example::
+
+            >>> print cube
+            air_temperature / (K)       (time: 12; latitude: 73; longitude: 96)
+                 Dimension coordinates:
+                      time                    x      -              -
+                      latitude                -      x              -
+                      longitude               -      -              x
+                 Auxiliary coordinates:
+                      year                    x      -              -
+            >>> cube.promote_aux_coord_to_dim_coord('year')
+            >>> print cube
+            air_temperature / (K)       (year: 12; latitude: 73; longitude: 96)
+                 Dimension coordinates:
+                      year                    x      -              -
+                      latitude                -      x              -
+                      longitude               -      -              x
+                 Auxiliary coordinates:
+                      time                    x      -              -
+
+        """
+
+        if isinstance(name_or_coord, basestring):
+            aux_coord = self.coord(name_or_coord)
+        elif isinstance(name_or_coord, iris.coords.Coord):
+            aux_coord = name_or_coord
+        else:
+            # Don't know how to handle this type
+            msg = ("Don't know how to handle coordinate of type {}. "
+                   "Ensure all coordinates are of type basestring or "
+                   "iris.coords.Coord.")
+            msg = msg.format(type(name_or_coord))
+            raise TypeError(msg)
+
+        if aux_coord in self.dim_coords:
+            # nothing to do
+            return
+
+        if not aux_coord in self.aux_coords:
```

Comment:
The recommended form for this is `if aux_coord not in self.aux_coords:`. This is what's causing the PEP8 checker to fail.